### PR TITLE
move sgrep to subdirectory, pin pfff as git submodule

### LINKED ISSUE #87: pin pfff version in Dockerfile
For reproducibility and debug-ability, we ought to be pinning the commit hash of pfff in the dockerfile. This will also make builds faster as we can move the `COPY` line down

### PR DESCRIPTION
**Pull request by ievans**
closes #87

### CODE REVIEW

**Comment by DrewDennison** [comment]
@ievans don't you want to keep the license docs etc all at the top level?

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,14 +10,9 @@ jobs:
...
       - checkout
-      - run: cd ~ && git clone  https://github.com/returntocorp/pfff
-      - run: eval $(opam env) && cd ~/pfff && ./configure && make depend && make && make opt && make install-libs
-      - run: eval $(opam env); make all
-      - run: cd sgrep_lint; make all
-      - persist_to_workspace:
-          root: /home/opam/project
-          paths:
-            - _build
+      - run: git submodule init && git submodule update --remote
```
Comment (DrewDennison) [inline .circleci/config.yml]: YAY!

**Comment by DrewDennison** [comment]
I don't see the reason to move stuff to sgrep folder. I would just leave as is but also bring in the pfff submodule

**Comment by ievans** [comment]
@DrewDennison dune complains if it sees pfff in a subfolder when doing a build from the root unless sgrep is in a separate folder. Perhaps @aryx knows another way, but I think this is cleaner: pfff, sgrep, sgrep_lint are all separate folders

**Comment by DrewDennison** [comment]
LGTM

**Pull request merged**